ui/overview_page.go

```diff
@@ -679,6 +679,8 @@ func (pg *overviewPage) Handler(eq event.Queue, c pageCommon) {
 			if e.Type == gesture.TypeClick {
 				txn := (*pg.walletTransactions).Recent[index]
 				*pg.walletTransaction = &txn
+				*c.returnPage = PageOverview
+				c.setReturnPage(PageOverview)
```

Comment:
This looks like the returnPage field is being set twice. It is set directly on line 682 once and a second time with a method 683.